packages/xrpl/src/utils/getNFTokenID.ts

```diff
@@ -15,19 +16,36 @@ interface NFToken {
   }
 }
 
+/**
+ * Ensures that the metadata is in a deserialized format to parse.
+ *
+ * @param meta - the metadata from a `tx` method call. Can be in json format or binary format.
+ * @returns the metadata in a deserialized format.
+ */
+function ensureDecodedMeta(
+  meta: TransactionMetadata | string | undefined,
+): TransactionMetadata | undefined {
+  if (typeof meta === 'string') {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Meta is either metadata or serialized metadata.
+    return decode(meta) as unknown as TransactionMetadata
+  }
+  return meta
+}
+
 /**
  * Gets the NFTokenID for an NFT recently minted with NFTokenMint.
  *
- * @param meta - Metadata from the response to submitting an NFTokenMint transaction.
+ * @param meta - Metadata from the response to submitting and waiting for an NFTokenMint transaction or from a `tx` method call.
  * @returns The NFTokenID for the minted NFT.
  * @throws if meta is not TransactionMetadata.
  */
 export default function getNFTokenID(
-  meta: TransactionMetadata,
+  meta: TransactionMetadata | string | undefined,
```

Comment:
Why is this allowed to be `undefined`? That seems erroneous.

Reply:
Agreed.  Undefined is strange.

Reply:
It's so that it can be used directly on the output of TxResponse.meta which is `TransactionMetadata | string | undefined` without needing custom logic to do typechecking/coercion. The `getNFTokenID` function will still throw if it receives undefined, but it'll do so with a consistent error instead of requiring any users of the function to write their own each time.

Reply:
IMO it's easy enough to do an `undefined` check.

Reply:
I feel like if you have to write the same error handler for every instance, it should just go in the function instead.

Especially since it's not obvious that the reason meta can be undefined is based on the transaction type you're looking up. So if they're searching for an NFTokenMint transaction, it should never be undefined. If they try searching up on xrpl.org how they should handle the undefined case for `meta` it's hard to get a clear answer using our docs as to what to do about it.

Reply:
Hmmm... Looking into it again I think you're right that it won't ever be undefined. 

It seems [way back when we did xrpl.js 2.0 I changed the type to be optional](https://github.com/XRPLF/xrpl.js/pull/1636/files#), but I'm guessing I did that because I confused the `tx` request with the `tx` field in a submitted Transaction. 

![image](https://github.com/XRPLF/xrpl.js/assets/19694186/021a1219-513d-48ab-b027-34a96c5fc927)

In a submitted transaction, the `meta` field is often undefined because the actual metadata is included as an additional field at the end. For a `tx` request though it seems that every transaction changes something on the ledger, so every `tx` response should have `meta` if it succeeds. 

Based on that, I think the best thing to do is to make `meta` not optional from a TxResponse, and then update this function to be the same (although I'm a bit worried that'd be a breaking change for TypeScript users because it'll give them errors of "could not be undefined" or something like that)

Reply:
Note: the "raw" view on the Explorer isn't actually the raw transaction, there's some minor mutations. So if you want to know what the raw transaction output looks like you'll need to look at the actual `tx` output.

Reply:
Bump on making this change to remove the `undefined` option